server/src/main/java/org/apache/iotdb/db/engine/settle/SettleRequestHandler.java

```diff
@@ -0,0 +1,321 @@
+        return RpcUtils.getStatus(
+            TSStatusCode.UNSUPPORTED_OPERATION,
+            "Too many files offered, the limited count of system config is "
+                + maxInnerCompactionCandidateFileNum
+                + ", the input file count is "
+                + tsFileNames.size());
+      }
+
+      TSStatus validationResult;
+      Pair<ConsistentSettleInfo, TSStatus> consistentSettleInfoStatusPair;
+
+      for (String path : paths) {
+        File currentTsFile = new File(path);
+        if (!currentTsFile.exists()) {
+          return RpcUtils.getStatus(
+              TSStatusCode.PATH_NOT_EXIST, "The specified file does not exist in " + path);
+        }
+        File modsFile = new File(path + ModificationFile.FILE_SUFFIX);
+        hasModsFiles |= modsFile.exists();
+
+        consistentSettleInfoStatusPair = calculateConsistentInfo(currentTsFile);
+        ConsistentSettleInfo currentInfo = consistentSettleInfoStatusPair.left;
+        if (this.consistentSettleInfo == null) {
+          this.consistentSettleInfo = currentInfo;
+        }
+        validationResult = consistentSettleInfoStatusPair.right;
+        if (!isSuccess(validationResult)) {
+          return validationResult;
+        }
+
+        validationResult = consistentSettleInfo.checkConsistency(currentInfo);
+        if (!isSuccess(validationResult)) {
+          return validationResult;
+        }
+
+        if (TsFileUtils.isSequence(currentTsFile)) {
+          hasSeqFiles = true;
+        } else {
+          hasUnseqFiles = true;
+        }
+        tsFileNames.add(currentTsFile.getName());
+        if (hasSeqFiles && hasUnseqFiles) {
+          return RpcUtils.getStatus(
+              TSStatusCode.UNSUPPORTED_OPERATION, "Settle by cross compaction is not allowed.");
+        }
+      }
+
+      if (!hasModsFiles) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PARAMETER,
+            "Every selected TsFile does not contains the mods file.");
+      }
+      DataRegion dataRegion =
+          StorageEngine.getInstance()
+              .getDataRegion(new DataRegionId(consistentSettleInfo.dataRegionId));
+      if (dataRegion == null) {
+        return RpcUtils.getStatus(TSStatusCode.ILLEGAL_PATH, "DataRegion not exist");
+      }
+      tsFileManager = dataRegion.getTsFileManager();
+
+      validationResult = checkCompactionConfigs();
+      if (!isSuccess(validationResult)) {
+        return validationResult;
+      }
+
+      if (hasSeqFiles) {
+        allTsFileResourceList =
+            tsFileManager.getSequenceListByTimePartition(consistentSettleInfo.timePartitionId);
+      } else {
+        allTsFileResourceList =
+            tsFileManager.getUnsequenceListByTimePartition(consistentSettleInfo.timePartitionId);
+      }
+
+      return validateTsFileResources();
+    }
+
+    private Pair<ConsistentSettleInfo, TSStatus> calculateConsistentInfo(File tsFile) {
```

Comment:
We need to avoid using `Pair` because it will add the complexity of codes.
We can put the `status` as a variable of ConsistentSettleInfo. 

What you want to do here is check whether the `info` can be totally got from given TsFile path or not. It not, the whole procedure should terminated.

So we can add a new variable such as `isValid` into `ConsistentSettleInfo` to represent whether the Info is ok or not